### DISCUSSION TITLE
Alarm 2 or Alarm1 when mist OFF (M9) [resolved]

### DISCUSSION THREAD

**Discussion by rmaltais** [grblHAL/core#507] [General]
Hi,
On my last firmware, i use a mist on program and when i turn off (M9), before go to tool setter, i was Alarm2 or Alarm1 on IOSender. No physical trigger about my end axis. When i remove all M9 in my program, i have no problem and program run normally. Strange think...

**Comment by terjeio** [comment]
EMI? If you disconnect the mist relay/solenoid does it still happen?

**Reply by rmaltais** [reply]
I'll investigate about that. thank

**Reply by rmaltais** [reply]
When i disconnect my solenoid, nothing happen, no alarm on close mist. No alarm. Just when close it with connected, i have always Alarm 1. 
When i disable hard limit, no alarm on close mist with solenoid connected.

**Reply by terjeio** [reply]
Ok, then back EMF from the solenoid is likely the cause.
How is it driven? Via a transistor? Is there a snubber diode across the coil?

**Reply by rmaltais** [reply]
Hi

Here is the schematic of the card which indicates how my solenoid is driven via EL357N and ULN2003. I don't have a diode installed between the terminals

![image](https://github.com/grblHAL/core/assets/41405404/f0039737-ee20-4dc4-85a2-3e0b1f0f8840)

Thank

**Reply by rmaltais** [reply]
As you suggested, I installed a diode in parallel to my coil and the problem is fixed.
Thank you again for your help.

**Comment by rmaltais** [comment]
> Ok, then back EMF from the solenoid is likely the cause. How is it driven? Via a transistor? Is there a snubber diode across the coil?
